commands/update.py

```diff
@@ -61,7 +64,9 @@ def do_update(args):
             url = urlutil.get_s3_url(s3_path)
             new_version = args.commit_hash
         else:
-            logging.critical(" we didn't find any firmware links with commit hash '{}'".format(args.commit_hash))
+            message = "We didn't find any firmware links with commit hash '{}'".format(args.commit_hash)
+            logging.critical(message)
+            raise UpdateError(message)
```

Comment:
At this point we will not be going to space, so fine, be cranky about it. I'll make a top-level exception handler at some point. But I think pretty much any time we have a `critical` observation the world has ended, so we should stop.